### PR TITLE
Fix golint issues in some pkg/controller packages

### PR DESCRIPTION
**Pull request by PrasadG193**
**What this PR does / why we need it**:
Fixes golint issues in:
-pkg/controller/clusterroleaggregation
-pkg/controller/cronjob
-pkg/controller/daemon
-pkg/controller/deployment
-pkg/controller/deployment/util
-pkg/controller/disruption
-pkg/controller/endpoint

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes part of  #68026

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by PrasadG193** [comment]
Due to "no stuttering' rule in golint and for the sake of uniformity, `ClusterRoleAggregationController`, `DaemonSetsController`, `DeploymentController`, `CronJobController`, `EndpointController`, `DisruptionController` are renamed into `Controller`

**Comment by PrasadG193** [comment]
Hey @cheftako @deads2k, could you please review the PR and let me know if any changes needed

File: pkg/controller/cronjob/cronjob_controller.go [hunk 1]
```diff
@@ -58,15 +58,17 @@ import (
 // controllerKind contains the schema.GroupVersionKind for this controller type.
 var controllerKind = batchv1beta1.SchemeGroupVersion.WithKind("CronJob")
 
-type CronJobController struct {
+// Controller is a controller to manage cronjobs
```
Comment (cheftako) [inline pkg/controller/cronjob/cronjob_controller.go]: I like the removal of comment redundancy above. Can we make this ?
// Controller to manage cronjobs

File: pkg/controller/cronjob/utils.go [hunk 2]
```diff
@@ -145,7 +145,7 @@ func getRecentUnmetScheduleTimes(sj batchv1beta1.CronJob, now time.Time) ([]time
 		// but less than "lots".
 		if len(starts) > 100 {
 			// We can't get the most recent times so just return an empty slice
-			return []time.Time{}, fmt.Errorf("Too many missed start time (> 100). Set or decrease .spec.startingDeadlineSeconds or check clock skew.")
+			return []time.Time{}, fmt.Errorf("Too many missed start time (> 100). Set or decrease .spec.startingDeadlineSeconds or check clock skew")
```
Comment (cheftako) [inline pkg/controller/cronjob/utils.go]: Sorry not your fault but if we are fixing this line, errors are supposed to start with a lower case character.

File: pkg/controller/endpoint/endpoints_controller.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ const (
 	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
 	maxRetries = 15
 
-	// An annotation on the Service denoting if the endpoints controller should
+	// TolerateUnreadyEndpointsAnnotation annotation on the Service denotes if the endpoints controller should
```
Comment (cheftako) [inline pkg/controller/endpoint/endpoints_controller.go]: Can we fix the redundancy? (Drop the "annotation"?)

File: pkg/controller/endpoint/endpoints_controller.go [hunk 4]
```diff
@@ -68,13 +68,13 @@ const (
 	TolerateUnreadyEndpointsAnnotation = "service.alpha.kubernetes.io/tolerate-unready-endpoints"
 )
 
-// NewEndpointController returns a new *EndpointController.
-func NewEndpointController(podInformer coreinformers.PodInformer, serviceInformer coreinformers.ServiceInformer,
-	endpointsInformer coreinformers.EndpointsInformer, client clientset.Interface) *EndpointController {
+// NewController returns a new *Controller.
```
Comment (cheftako) [inline pkg/controller/endpoint/endpoints_controller.go]: for reconciling endpoints?

**Review by cheftako — COMMENTED** [review]
A couple of minor comments.

**Comment by PrasadG193** [comment]
@cheftako I have addressed review comments and squashed commits

**Review by cheftako — APPROVED** [review]
/lgtm

**Comment by PrasadG193** [comment]
@cheftako can you please add `ok-to-test` label?

**Comment by PrasadG193** [comment]
@cheftako I have fixed the test case failures. Can you please add `lgtm` label again?

File: pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go [hunk 5]
```diff
@@ -49,8 +49,8 @@ type ClusterRoleAggregationController struct {
 }
 
 // NewClusterRoleAggregation creates a new controller
-func NewClusterRoleAggregation(clusterRoleInformer rbacinformers.ClusterRoleInformer, clusterRoleClient rbacclient.ClusterRolesGetter) *ClusterRoleAggregationController {
-	c := &ClusterRoleAggregationController{
+func NewClusterRoleAggregation(clusterRoleInformer rbacinformers.ClusterRoleInformer, clusterRoleClient rbacclient.ClusterRolesGetter) *Controller {
```
Comment (cheftako) [inline pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go]: Should consider making this NewController for consistency.

**Review by cheftako — APPROVED** [review]
/lgtm

File: pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go (hunk 5 above)
Comment (PrasadG193) [reply pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go]: Hmm, sounds good. Will do the changes.

Comment (PrasadG193) [reply pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go]: @cheftako Done.

**Comment by PrasadG193** [comment]
Looks like `lgtm` label is removed after rebasing. Can someone please add `lgtm`?
@cheftako

**Comment by dims** [comment]
@PrasadG193 please rebase

**Comment by dims** [comment]
@PrasadG193 can you please look into the verify failure?

**Comment by PrasadG193** [comment]
@dims golint is failing for auto-generated code (like https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/config/v1alpha1/conversion.go). Not sure how we are handling this in other packages

**Comment by dims** [comment]
@PrasadG193 can you add it to https://github.com/kubernetes/kubernetes/blob/master/hack/.golint_failures ?

**Comment by PrasadG193** [comment]
Can someone please add `/lgtm`?

**Review by cheftako — COMMENTED** [review]
/lgtm

**Comment by oomichi** [comment]
@PrasadG193 Could you rebase this for the latest master branch?
The test part seems good for me anyways.

/approve

**Comment by danwinship** [comment]
@PrasadG193 needs rebase. Also, you might have better luck getting this reviewed in pieces scoped to individual components so you don't need multiple approvals to merge

**Comment by PrasadG193** [comment]
Will do. I'll split the PR if too much refactoring needed :slightly_smiling_face:

**Comment by PrasadG193** [comment]
Rebased with master

**Comment by PrasadG193** [comment]
@cheftako @bowei Please review

**Pull request closed without merging**